src/explore_service.py

```diff
@@ -500,16 +472,30 @@ def explore_kafka(self, service_name, project):
     
     # If the service has Kafka connect, we can explore it as well
     if(kafka["user_config"]["kafka_connect"]==True):
-        (newnodes, newedges) = explore_kafka_connect(self, service_name, project, "kafka")
+        _, _, newnodes, newedges = explore_kafka_connect(self, None, service_name, project, "kafka")
```

Comment:
This is failing my side with the following error

```
Error looking up data for RUNNING kafka service demo-kafka: TypeError explore_kafka_connect() takes 4 positional arguments but 5 were given
```

Reply:
Apologies for missing that!

`explore_kafka_connect` is unusual in that it is called from the main "explore" part to explore a Kafka Connect node, but is also called from `explore_kafka`

The extra `service` and the extra return values are to make it fit the interface for the methods that are called from the main "explore" part (they take the `service` because some of them were recalculating it, and the caller could pass it down, and they return the host/port because that wasn't the same for all services, and moving it into the methods meant doing an "explore" on a node was just one method call).

The left-over "kafka" argument is from the earlier code - the `explore_kafka` method passed it to say "you're being called from `explore_kafka`, so need to behave a bit differently". I should have removed it, because now we're passing in the `service` value, we can use it being None to distinguish this case.

I believe I've fixed the incorrect call now.

Reply:
All works!